### ISSUE TITLE
Subsetting tibbles downgrades `labelled_spss` elements to `labelled` elements

### ISSUE THREAD

**Issue by huftis** [tidyverse/haven#360]
I’m not sure this is a `haven` or a `tibble` bug, but the behaviour seems somewhat `haven`-specific, so I’m reporting it here. When subsetting a tibble containing a `labelled_spss` element, the element seems to be downgraded to just a `labelled` object. 

More specifically, the (multi-)class `'labelled_spss', 'labelled'` is changed to just `'labelled'` and the `na_values`, `na_range` and `label` attributes are discarded (while the `labels` attribute is retained). Reprex:

```r
library(haven)
library(tibble)
gen = labelled_spss(c(0, 0, 1, 99), c(Female = 0, Male = 1),
                    na_values = 99, na_range = c(990, 1000))
attr(gen, "label") = "Patient gender"

d = tibble(gen)
str(d$gen) # All metadata
#> Classes 'labelled_spss', 'labelled'  atomic [1:4] 0 0 1 99
#>   ..- attr(*, "labels")= Named num [1:2] 0 1
#>   .. ..- attr(*, "names")= chr [1:2] "Female" "Male"
#>   ..- attr(*, "na_values")= num 99
#>   ..- attr(*, "na_range")= num [1:2] 990 1000
#>   ..- attr(*, "label")= chr "Patient gender"
str(d[1:4,]$gen) # Only some metadata
#> Class 'labelled'  atomic [1:4] 0 0 1 99
#>   ..- attr(*, "labels")= Named num [1:2] 0 1
#>   .. ..- attr(*, "names")= chr [1:2] "Female" "Male"
```

(BTW, the reason I create the `label` attribute manually, is that the `labelled_spss()` (and `labelled()`) generator function don’t have a mechanism for creating it. However, the `haven` package *does* support the attribute. For example, reading SPSS files with `read_spss()` automatically creates the attribute if a variable has a label. And it’s a very useful attribute to have, so it shouldn’t be discarded.)

**Comment by huftis** [comment]
Having taken a closer look at this issue, I think it’s related to the `[.labelled` method (and the corresponding *missing* `[.labelled_spss` method).

**Comment by huftis** [comment]
Looks like this will be fixed by #353.
